spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/health/PubSubHealthIndicator.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.pubsub.health;
+
+import java.util.UUID;
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.StatusCode;
+
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
+
+/**
+ * Default implemenation of
+ * {@link org.springframework.boot.actuate.health.HealthIndicator} for Pub/Sub. Validates
+ * if connection is successful by looking for a random generated subscription name that
+ * won't be found. If no subscription is found we know the client is able to connect to
+ * GCP Pub/Sub APIs.
+ *
+ * @author Vinicius Carvalho
+ *
+ * @since 1.2
```

Comment:
We're on 1.3 now. :-)
```suggestion
 * @since 1.3
```